extensions/azure-functions/deployment/src/main/java/io/quarkus/azure/functions/deployment/AzureFunctionsProcessor.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.azure.functions.deployment;
+
+import java.lang.reflect.Modifier;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.arc.processor.BuiltinScope;
+import io.quarkus.deployment.Feature;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.pkg.PackageConfig;
+import io.quarkus.deployment.pkg.builditem.LegacyJarRequiredBuildItem;
+
+public class AzureFunctionsProcessor {
+    private static final Logger log = Logger.getLogger(AzureFunctionsProcessor.class);
+
+    @BuildStep
+    public LegacyJarRequiredBuildItem forceLegacy(PackageConfig config) {
+        // Azure Functions need a legacy jar and no runner
+        config.addRunnerSuffix = false;
```

Comment:
What's the deal here? Why are we changing the config? This is generally something we try to avoid as much as possible by using/introducing a BuildItem

Reply:
I really dislike all the BuildItems surrounding the override of PackageConfig.  Have you seen the logic to check these types of overriding build items?  Its a mess.  This is much simpler.

Reply:
I'd prefer a multi build item called "OverridePackageConfigBuildItem".  steps that change PackageConfig just modify PackageConfig directly and produce that item.  Steps that need finished PackageConfig just order themselves by consuming both OverridePackageConfigBuildItem and PackageConfig.  

Again, the current way we do this is a mess and I had no desire to make it worse.

Reply:
The problem with changing config value in build steps is that there is hard guarantee about when it will happen

Reply:
And that's a problem because theoretically there could be build steps that read the value and perform some action.
If the value is not immutable, you could get inconsistent output

Reply:
Do a search for LegacyJarRequiredBuildItem you'll see how ugly things are.

PackageConfig is already indirectly mutable by these build items like LegacyJarRequiredBuildItem.  There's a few of these types of build items and the logic is messy.  So extension code already could get inconsistent if it is not aware of them.  

The build step in my PR produces LegacyJarRequiredBuildItem so correct ordering is already enforced so long as extensions interested in PackageConfig consume that build item (There are a few other build items like this too!!!)

Again, I think a better solution is to have a "OverridePackageConfigBuildItem" and modify PackageConfig directly.  But I'd rather do it in a separate PR.

Reply:
Your point is valid (as you are using the build item as well), but I am very reluctant to have an instance of this pattern in the core, as then it would give others a way to argue of doing it in their other PRs.

Reply:
Things are already potentially broken in main as these processors look at PackageConfig, but do not pull in items like LegacyJarRequiredBuildItem:
JibProcessor
S2iProcessor
OpenShiftProcessor

Luckily I think these overrides for PackageConfig only come into play when deploying to azure functions and lambda :)

Also look at JarResultBuildStep and how much more simplified the code would be if PackageConfig could be just modified and used directly instead of checking for the existence of the build items too.  When you start doing fine grain override of config like I'm doing in this PR, this code in JarResultBuildStep will become much much worse.

Reply:
Okay, let's do it that way but please follow up with the cleanup mentioned so we can have a uniform package config access pattern everywhere (and consider it the only exception to the immutability of config which we allow due to the complexity of the specific use case)